lib/replicache.types.ts

```diff
@@ -0,0 +1,50 @@
+import type { MutationV1 } from 'replicache'
+import { z } from 'zod'
+
+export type ClientPushInfo = {
+  client_id: MutationV1['clientID']
+  last_mutation_id_in_db: MutationV1['id']
+  last_mutation_id_in_request: MutationV1['id']
+}
+
+export const CustomPullRequest = z.object({
+  cookie: z.union([z.number(), z.null()]),
```

Comment:
```suggestion
  cookie: z.number().nullable(),
```

Gives better error messages. `z.union` gives a pretty complicated error message since it gives you one for each member of the union that failed.

Reply:
I've switched the Replicache strategy and made use of `z.nullable`, which I assume yields the same results?